spec/spec_helper.rb

```diff
@@ -114,11 +115,23 @@ class Pet < ActiveRecord::Base
   col :home_address, :type => :text
   if ENV['DB'] == 'postgresql'
     col :tsntz, :type => 'timestamp without time zone'
+  else
+    add_index :name, :unique => true
+  end
+end
+if ENV['DB'] == 'postgresql'
+  begin
+    Pet.connection.execute("ALTER TABLE pets DROP CONSTRAINT IF EXISTS unique_name")
+  rescue => e
+    puts e.inspect
   end
-  add_index :name, :unique => true
 end
+
 Pet.auto_upgrade!
 
+if ENV['DB'] == 'postgresql'
+  Pet.connection.execute("ALTER TABLE pets ADD CONSTRAINT unique_name UNIQUE (name)")
```

Comment:
this is the main different between native-upsert and our custom upsert functions (on postgres): the former requires a unique index to work